go.mod

```diff
@@ -58,64 +58,69 @@ require (
 	github.com/spf13/cobra v1.4.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.1
+	github.com/vmware/go-vcloud-director/v2 v2.15.0
 	github.com/vmware/govmomi v0.28.0
 	go.anx.io/go-anxcloud v0.4.3
 	go.etcd.io/etcd/api/v3 v3.5.4
 	go.etcd.io/etcd/client/pkg/v3 v3.5.4
 	go.etcd.io/etcd/client/v3 v3.5.4
 	go.etcd.io/etcd/etcdutl/v3 v3.5.4
 	go.uber.org/zap v1.21.0
-	golang.org/x/crypto v0.0.0-20220507011949-2cf3adece122
-	golang.org/x/oauth2 v0.0.0-20220411215720-9780585627b5
+	golang.org/x/crypto v0.0.0-20220525230936-793ad666bf5e
+	golang.org/x/oauth2 v0.0.0-20220524215830-622c5d57e401
 	golang.org/x/tools v0.1.10
 	gomodules.xyz/jsonpatch/v2 v2.2.0
 	google.golang.org/api v0.77.0
-	google.golang.org/genproto v0.0.0-20220421151946-72621c1f0bd3
-	google.golang.org/grpc v1.45.0
+	google.golang.org/genproto v0.0.0-20220531134929-86cf59382f1b
+	google.golang.org/grpc v1.46.2
 	google.golang.org/protobuf v1.28.0
 	gopkg.in/fsnotify.v1 v1.4.7
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/square/go-jose.v2 v2.6.0
 	gopkg.in/yaml.v2 v2.4.0
-	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
-	k8c.io/kubeone v1.4.3
+	gopkg.in/yaml.v3 v3.0.1
+	k8c.io/kubeone v1.4.1-0.20220531113023-c1f699725c2a
 	k8c.io/operating-system-manager v0.4.2
-	k8s.io/api v0.23.6
-	k8s.io/apiextensions-apiserver v0.23.6
-	k8s.io/apimachinery v0.23.6
-	k8s.io/apiserver v0.23.6
+	k8s.io/api v0.24.1
+	k8s.io/apiextensions-apiserver v0.24.1
+	k8s.io/apimachinery v0.24.1
+	k8s.io/apiserver v0.24.1
 	k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0
-	k8s.io/cli-runtime v0.23.6
+	k8s.io/cli-runtime v0.24.1
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/code-generator v0.23.6
-	k8s.io/klog/v2 v2.40.1
-	k8s.io/kube-aggregator v0.23.6
-	k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf
-	k8s.io/kubectl v0.23.6
-	k8s.io/metrics v0.23.6
+	k8s.io/code-generator v0.24.0
+	k8s.io/klog/v2 v2.60.1
+	k8s.io/kube-aggregator v0.24.1
+	k8s.io/kube-openapi v0.0.0-20220413171646-5e7f5fdc6da6
+	k8s.io/kubectl v0.24.1
+	k8s.io/metrics v0.24.0
 	k8s.io/test-infra v0.0.0-20210826180422-39483c498f0f
-	k8s.io/utils v0.0.0-20211116205334-6203023598ed
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
 	kubevirt.io/api v0.53.0
-	kubevirt.io/containerized-data-importer-api v1.48.1
+	kubevirt.io/containerized-data-importer-api v1.49.0
 	sigs.k8s.io/aws-iam-authenticator v0.5.8
-	sigs.k8s.io/controller-runtime v0.11.2
+	sigs.k8s.io/controller-runtime v0.12.1
 	sigs.k8s.io/controller-tools v0.8.0
 	sigs.k8s.io/yaml v1.3.0
 )
 
+// Added for testing purposes will be removed before merging this PR
+replace sigs.k8s.io/aws-iam-authenticator => github.com/ahmedwaleedmalik/aws-iam-authenticator v0.5.8-0.20220531123014-13bd59d5c7e9
```

Comment:
This will be removed before merging this PR. Against either of https://github.com/kubermatic/kubermatic/pull/9826 or https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/460, whichever gets merged first.